zhaquirks/tuya/ts0601_din_power.py

```diff
@@ -0,0 +1,171 @@
+TUYA_CURRENT_ATTR = 0x0212
+TUYA_POWER_ATTR = 0x0213
+TUYA_VOLTAGE_ATTR = 0x0214
+TUYA_DIN_SWITCH_ATTR = 0x0101
+
+SWITCH_EVENT = "switch_event"
+
+
+class TuyaManufClusterDinPower(TuyaManufClusterAttributes):
+    """Manufacturer Specific Cluster of the Tuya Power Meter device."""
+
+    manufacturer_attributes = {
+        TUYA_TOTAL_ENERGY_ATTR: ("energy", t.uint16_t),
+        TUYA_CURRENT_ATTR: ("current", t.int16s),
+        TUYA_POWER_ATTR: ("power", t.uint16_t),
+        TUYA_VOLTAGE_ATTR: ("voltage", t.uint16_t),
+        TUYA_DIN_SWITCH_ATTR: ("switch", t.uint8_t),
+    }
+
+    def _update_attribute(self, attrid, value):
+        super()._update_attribute(attrid, value)
+        if attrid == TUYA_TOTAL_ENERGY_ATTR:
+            self.endpoint.device.energy_bus.listener_event(
+                "energy_reported", value / 100
+            )
+        elif attrid == TUYA_CURRENT_ATTR:
+            self.endpoint.device.electrical_bus.listener_event(
+                "current_reported", value
+            )
+        elif attrid == TUYA_POWER_ATTR:
+            self.endpoint.device.electrical_bus.listener_event(
+                "power_reported", value / 10
+            )
+        elif attrid == TUYA_VOLTAGE_ATTR:
+            self.endpoint.device.electrical_bus.listener_event(
+                "voltage_reported", value / 10
+            )
+        elif attrid == TUYA_DIN_SWITCH_ATTR:
+            self.endpoint.device.switch_bus.listener_event(
+                "switch_event", attrid, value
+            )
+
+
+class TuyaPowerMeasurement(LocalDataCluster, ElectricalMeasurement):
+    """Custom class for power, voltage and current measurement."""
+
+    cluster_id = ElectricalMeasurement.cluster_id
+    POWER_ID = 0x050B
+    VOLTAGE_ID = 0x0505
+    CURRENT_ID = 0x0508
+
+    AC_CURRENT_MULTIPLIER = 0x0602
+    AC_CURRENT_DIVISOR = 0x0603
+
+    _CONSTANT_ATTRIBUTES = {AC_CURRENT_MULTIPLIER: 1, AC_CURRENT_DIVISOR: 1000}
+
+    def __init__(self, *args, **kwargs):
+        """Init."""
+        super().__init__(*args, **kwargs)
+
+        self.endpoint.device.electrical_bus.add_listener(self)
+
+    def voltage_reported(self, value):
+        """Voltage reported."""
+        self._update_attribute(self.VOLTAGE_ID, value)
+
+    def power_reported(self, value):
+        """Power reported."""
+        self._update_attribute(self.POWER_ID, value)
+
+    def current_reported(self, value):
+        """Ampers reported."""
+        self._update_attribute(self.CURRENT_ID, value)
+
+
+class TuyaElectricalMeasurement(LocalDataCluster, Metering):
+    """Custom class for total energy measurement."""
+
+    cluster_id = Metering.cluster_id
+    CURRENT_ID = 0x0000
+    POWER_WATT = 0x0000
+
+    """Setting unit of measurement."""
+    _CONSTANT_ATTRIBUTES = {0x0300: POWER_WATT}
+
+    def __init__(self, *args, **kwargs):
+        """Init."""
+        super().__init__(*args, **kwargs)
+        self.endpoint.device.energy_bus.add_listener(self)
```

Comment:
Do the same thing for the Metering cluster. Remove the `Bus` and update the values as outlined above.